packages/muelu/src/Graph/MatrixTransformation/MueLu_CoalesceDropFactory_def.hpp

```diff
@@ -572,109 +577,172 @@ void CoalesceDropFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::Build(Level
               }
               rows(row + 1) = realnnz;
             }
-          } else {
-            /* Cut Algorithm */
-            // CMS
-            using DropTol = Details::DropTol<real_type, LO>;
-            std::vector<DropTol> drop_vec;
-            drop_vec.reserve(nnz);
-            const real_type zero = Teuchos::ScalarTraits<real_type>::zero();
-            const real_type one  = Teuchos::ScalarTraits<real_type>::one();
-            LO rownnz            = 0;
-            // NOTE: This probably needs to be fixed for rowsum
-
-            // find magnitudes
-            for (LO colID = 0; colID < (LO)nnz; colID++) {
-              LO col = indices[colID];
-              if (row == col) {
-                drop_vec.emplace_back(zero, one, colID, false);
-                continue;
+          }  // end for row
+        }
+        else {
+          /* Cut Algorithm */
+          SubFactoryMonitor m1(*this, "Cut Drop", currentLevel);
+          using ExecSpace = typename Node::execution_space;
+          using TeamPol = Kokkos::TeamPolicy<ExecSpace>;
+          using TeamMem = typename TeamPol::member_type;
+          using ATS = Kokkos::ArithTraits<Scalar>;
+          using impl_scalar_type = typename ATS::val_type;
+          using implATS = Kokkos::ArithTraits<impl_scalar_type>;
+
+          //move from host to device
+          auto ghostedDiagValsView = Kokkos::subview(ghostedDiag->getDeviceLocalView(Xpetra::Access::ReadOnly), Kokkos::ALL(), 0);
+          auto boundaryNodesDevice = Kokkos::create_mirror_view(ExecSpace(), boundaryNodes);
```

Comment:
Don't we need to deep_copy that from host to device?

Reply:
You're right. I found the same issue on line 734 and line 739.